lib/konfa.rb

```diff
@@ -63,16 +60,27 @@ def on_key_missing(key, value)
         raise UnsupportedVariableError.new(key)
       end
 
-      def do_deferred_initialization?
-        !self.initialized_deferred && !self.deferred.nil?
-      end
-
       #
       # The following methods provides the interface to this class
       #
 
+      def init
+        return if self.initialized == true
+        # Set to true before calling to prevent recursion if
+        # an initializer is accessing the configuration
+        self.initialized = true
+        unless self.initializer.nil?
+          self.send(self.initializer.first, *self.initializer[1..-1])
+        end
+        self.after_initialize
+      end
+
+      def reset_configuration
```

Comment:
This should be called "reinit" or "reset" or something. For the sake of the API, it doesn't actually reset the configuration, it will just run init again next time a variable is accessed.